xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/js/xwiki/viewers/attachments.js

```diff
@@ -181,17 +181,19 @@ require(['jquery', 'xwiki-events-bridge'], function($) {
    * Getting the button that triggers the modal.
    */
   $(document).on('show.bs.modal', '#deleteAttachment', function(event) {
-    $(this).data('relatedTarget', $(event.relatedTarget));
+    if (event.relatedTarget) {
+      $(this).data('relatedTarget', $(event.relatedTarget));
+    }
   });
   /**
    * Event on deleteAttachment button.
    */
   $(document).on('click', '#deleteAttachment input.btn-danger', function() {
     var modal = $('#deleteAttachment');
-    var button = modal.data('relatedTarget');
+    var button = $(modal.data('relatedTarget'));
```

Comment:
Are you sure you need to wrap with jQuery? I see above that you set ``$(event.relatedTarget)``.